isisd/isis_sr.c

```diff
@@ -1603,8 +1637,7 @@ static void sr_adj_sid_add_single(struct isis_adjacency *adj, int family,
  */
 static void sr_adj_sid_add(struct isis_adjacency *adj, int family)
 {
-	sr_adj_sid_add_single(adj, family, false);
-	sr_adj_sid_add_single(adj, family, true);
+	sr_adj_sid_add_single(adj, family, false, NULL);
```

Comment:
You must leave the call to `sr_adj_sid_add_single(adj, family, true);` for the backup adjacency SID even if TI-LFA is not enable. Indeed, backup Adjacency SID is also used for any link protection like MPLS FRR. You must add a call to `sr_adj_sid_add_single(adj, family, true, NULL);`

Reply:
Agreed. The "backup" flag means that the Adj-SID is eligible for protection, not that it's necessarily protected. So advertising backup Adj-SIDs even when TI-LFA isn't configured is the ideal thing to do (that's what other routing stacks are doing at least).

As we discussed a while ago, I'll defer that work to a later time since it's a change that will require a lot of manual testing (unfortunately our topotests don't test Adj-SIDs as of now because of their dynamic nature).

Reply:
I think it is not too much work to just add a call to `sr_adj_sid_add_single(adj, family, true, NULL);` This will not break TI-LFA if you just look to the non backup Adjacency SID.

But agree that we could differ this change for another PR. But, we don't forget to do that. I'll add it to our project task list